modules/loot-table/src/main/java/unnamed/mmo/loot/impl/LootContextImpl.java

```diff
@@ -0,0 +1,33 @@
+package unnamed.mmo.loot.impl;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import unnamed.mmo.data.NumberSource;
+import unnamed.mmo.loot.LootContext;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public record LootContextImpl(
+        @NotNull Map<String, Object> dataMap,
```

Comment:
Using a string - object map is a bit odd for me, can this be replaced with a generic?